xarray/tests/test_namedarray.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import pytest
+
+from xarray.namedarray.core import NamedArray, as_compatible_data
+from xarray.namedarray.utils import T_DuckArray
+
+
+@pytest.fixture
+def random_inputs():
+    return np.random.random((3, 4, 5)).astype(np.float32)
```

Comment:
this tends to be hard to debug.
```suggestion
    return np.arange(3*4*5).reshape((3, 4, 5)).astype(np.float32)
```